### PR TITLE
Update etcd txn API doc to clarify the inner revision in TxnResponse

### PR DESCRIPTION
**Pull request by ahrtr**
Clarify https://github.com/etcd-io/etcd/issues/14490 

Signed-off-by: Benjamin Wang <wachao@vmware.com>

cc @spzala @serathius @ptabor

### CODE REVIEW

File: content/en/docs/v3.6/learning/api.md [hunk 1]
```diff
@@ -323,6 +323,10 @@ message ResponseOp {
 }
 ```
 
+For `response_range` and `response_delete_range` responses in `TxnResponse`, the `revision` included in the inner response header is just the revision when generating the response, so it might be one less than the `revision` included
```
Comment (socketpair) [inline content/en/docs/v3.6/learning/api.md:326]: Actually all responses, without clarification. And also without clarification about "one less".

I would write something like:

```
Response headers returned in TXN should not be interpreted in any way, and should be completely ignored.
```

Comment (ahrtr) [reply content/en/docs/v3.6/learning/api.md]: Thanks for the feedback. Just updated the doc. 

It isn't decided to deprecate the inner `ResponseHeader` for now. Please note that each response has exactly the same format as non-TXN response. 

The `revision` included in each response might be still useful, such as checking whether the deletion is successful or not when there is no PUT request before it.

Comment (ahrtr) [reply content/en/docs/v3.6/learning/api.md]: Makes sense to me & updated & thx.

Note that there is no plan to remove the `ResponseHeader` in each inner response. See https://github.com/etcd-io/etcd/issues/14490#issuecomment-1257086651

**Comment by ahrtr** [comment]
@spzala @serathius Please take a look.

**Comment by vm86** [comment]
I'm writing on behalf of @socketpair :) 
LGTM!

**Comment by socketpair** [comment]
Agreed

**Review by spzala — APPROVED** [review]
Good clarification. 
lgtm Thanks @ahrtr @socketpair

**Comment by ahrtr** [comment]
@spzala @serathius Please help to merge this PR if there is no any objection. thx

**Comment by spzala** [comment]
Sure @ahrtr

**Pull request merged**